Sources/SwiftUINavigation/Overlay.swift

```diff
@@ -0,0 +1,55 @@
+
+extension View {
+    /// Presents an overlay using a bind as a data source for the overlay's content.
+    ///
+    /// - Parameters:
+    ///   - value: A binding to a source of truth for the overlay. When `value` is non-`nil`, a
+    ///     non-optional binding to the value is passed to the `content` closure. You use this binding
+    ///     to produce content that the system presents to the user in an overlay. Changes made to the
+    ///     overlay's binding will be reflected back in the source or truth. Likewise, changes to
+    ///     `value` are instantly reflected in the modal. If `value` becomes `nil`, the overlay is
+    ///     dismissed.
+    ///   - onDismiss: The closure to execute when dismissing the overlay.
+    ///   - content: A closure returning the content of the modal.
+    public func overlay<Value, Content>(
+      unwrapping value: Binding<Value?>,
+      onDismiss: (() -> Void)? = nil,
```

Comment:
This doesn't appear to be used in the body of the function...should it get attached to a `onDisappear` somewhere?